### ISSUE TITLE
SVG icons not visible in WHCM

### ISSUE THREAD

**Issue by danchamp** [localgovdrupal/localgov_base#112]
[Accessibility audit: issue 76](https://docs.google.com/spreadsheets/d/1t8rRIQe_1kogxPBZDKcsbf8YxtEfC9iGJvidkagNSXY/edit#gid=1779042480&range=E77)

SVG fill and stroke properties are not over-ridden by Window High Contrast Mode, and most localgov_base SVGs have near-black fill and stroke, so aren't visible.

Can target WHCM with a media query:

`@media screen and (-ms-high-contrast: active)`

**Comment by waako** [comment]
We need to ensure we are using the `force-colors` media query also, as `-ms-high-contrast` is being retired.

[Adrian Roseli has an article about WHCM and system colours](https://adrianroselli.com/2021/02/whcm-and-system-colors.html), with some specific [examples around SVG](https://adrianroselli.com/2021/02/whcm-and-system-colors.html#SVG). [Sarah Higley's quick tips for WHCM](https://sarahmhigley.com/writing/whcm-quick-tips/) are a good base also.

**Comment by waako** [comment]
Actually, it would be much more sensible to provide a default `fill: currentColor`, and only resort to media queries where necessary.

For example, the [Adult Health & social care](https://demo.localgovdrupal.org/adult-health-and-social-care) contact info section icons, screenshots where `fill: currentColor` added to `.service-landing-page__contact-icon { `:

- Default with the fill set:  
  <img width="559" alt="Contact info section with icons have fill property set to currentColor" src="https://github.com/localgovdrupal/localgov_base/assets/70727/e8ecb055-3671-40bd-b289-accc4e5782fd">
- Windows Hight Contrast mode, icons now visible:  
  <img width="441" alt="Contact info section with icons have fill property set to currentColor, now visible in Windows High Contrast Mode" src="https://github.com/localgovdrupal/localgov_base/assets/70727/0b5b2723-981f-47c6-b51b-914be680934e">

For reference, Contact Info section icons currently in Windows High Contrast mode:
<img width="471" alt="LGD-contact-info-icons-no-fill-WHCM" src="https://github.com/localgovdrupal/localgov_base/assets/70727/eb1dcf26-e692-4ca0-91fb-2c889d668057">

**Comment by waako** [comment]
Need to check where in the theme this could be applied to have a safe but effective knock-on effect

**Comment by markconroy** [comment]
I'd like to get this fixed, but I'm starting to think time may have caught up with us.

Setting icon `fill: currentColor` seems fine in some places in LocalGov Base and LocalGov Scarfolk, but in other places the `currentColor` may be a colour we don't want to use. And we also don't want to force `currentColor` on lots of existing sites if they have already set background colours that might clash with us setting `currentColor`. 

I'm happy to look at a PR that uses `force-colors` and only targets WHCM. @waako do you want to have a go at that PR?

---
Thanks to [Big Blue Door](https://www.bigbluedoor.net/) for sponsoring my time to work on this.

**Comment by tonypaulbarker** [comment]
Thanks all. 

Moved to https://git.drupalcode.org/project/localgov_base/-/work_items/3590895
